R/utilities-simulation-results.R

```diff
@@ -223,59 +223,30 @@ simulationResultsToDataFrame <- function(simulationResults,
     values_to = "simulationValues"
   )
 
-  # Extract units and dimensions for paths in a separate data frame
-  # iterate over the list (in `.x`) using index (names for list elements)
-  # and apply function in `.f` to each element.
-  #
-  # The result will be a list of data frames, which will be row-wise glued into
-  # a single data frame with the `_dfr` variant of this function.
-  simMetaData <- purrr::imap_dfr(
-    .x = simList$metaData,
-    .f = ~ as.data.frame(.x, row.names = NULL, stringsAsFactors = FALSE),
-    .id = "paths"
-  )
-
-  # Leave out time units and dimensions since time is not a path; they will be
-  # added at a later stage.
-  simMetaData <- dplyr::filter(simMetaData, paths != "Time")
-
-  # Combine data frame with simulated data and meta data.
-  simData <- dplyr::left_join(simData, simMetaData, by = "paths")
-
-  # Add back in the previously left out time meta data to the combined data frame.
-  simData <- dplyr::bind_cols(
-    simData,
-    dplyr::tibble(
-      "TimeUnit" = simList$metaData$Time$unit[[1]],
-      "TimeDimension" = simList$metaData$Time$dimension[[1]]
+  units <- lapply(simList$metaData, function(path) {
+    path$unit
+  })
+  dims <- lapply(simList$metaData, function(path) {
+    path$dimension
+  })
+  molWeights <- lapply(unique(simData$paths), function(path) {
```

Comment:
@Felixmil any better way to do this?

Reply:
This should be better performance and readability wise:

```r
units <- purrr::map(simList$metaData, "unit")

dims <- purrr::map(simList$metaData, "dimension")

molWeights <- unique(simData$paths) %>%
  purrr::set_names() %>%
  purrr::map(~ospsuite::toUnit(
    quantityOrDimension = ospDimensions$`Molecular weight`,
    values              = simulationResults$simulation$molWeightFor(.x),
    targetUnit          = ospUnits$`Molecular weight`$`g/mol`
  ))
```
\+ remove `names(molWeights) <- unique(simData$paths)`

Reply:
Yepp a bit better.